builtin/providers/dockercloud/provider.go

```diff
@@ -0,0 +1,50 @@
+package dockercloud
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+// Provider returns a schema.Provider for DockerCloud
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"user": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("DOCKERCLOUD_USER", nil),
+				Description: "The user to authenticate as.",
+			},
+			"apikey": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("DOCKERCLOUD_APIKEY", nil),
+				Description: "API key used to authenticate the user.",
+			},
+			"baseurl": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
```

Comment:
If there is a default here, I believe this should be optional otherwise, the users will be promoted to add it all the time